### PR TITLE
Updates how libusgscsm is found

### PR DESCRIPTION
**Pull request by jlaura**
## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by AustinSanders — COMMENTED** [review]
glob() root_dir keyword is python 3.10+, see https://stackoverflow.com/questions/74113693/error-glob-got-an-unexpected-keyword-argument-root-dir for an alternative

File: knoten/__init__.py [hunk 1]
```diff
@@ -1,20 +1,26 @@
 import ctypes
 from ctypes.util import find_library
-from distutils import sysconfig
+from glob import glob
 import os
 import warnings
 
-from . import csm
-
-from csmapi import csmapi
```
Comment (acpaquette) [inline knoten/__init__.py:9]: I think these lines need to stay for passing tests, that or the imports in the failing tests need to change

Comment (jlaura) [reply knoten/__init__.py:9]: So, OS X is passing. It's this line on a linux system: https://github.com/DOI-USGS/knoten/pull/124/files#diff-9845ec9ebca5ff5318d96b72dea2063013cf4baf376e5a14c7f19a49051f8bdaR18

Where the globbed `results[-1]` are being grabbed. I bet that result is None.... I'll look more tomorrow AM.

Comment (jlaura) [reply knoten/__init__.py:9]: Resolving: The issue was that the glob was hard coded for OSX/Darwin. It is searching for `*.#.#.#.dylib`. That obviously fails on linux. I swapped the search pattern and put in checks for linux/Darwin OS.

I did not look to see what the windows installation naming looked like for usgscsm. That will need to be added in a PR as well.

**Comment by jlaura** [comment]
@acpaquette @AustinSanders  All passing and ready for a re-review. Issue was how hard coded the shared library find was. I generalized to linux / OS X.

**Review by chkim-usgs — APPROVED** [review]

**Review by AustinSanders — APPROVED** [review]

**Pull request merged**